hazelcast/src/hazelcast/client/connection/InSelector.cpp

```diff
@@ -66,7 +66,7 @@ namespace hazelcast {
                             sleepingSocket->receive(&wakeUpSignal, sizeof(int));
                         } else {
                             boost::shared_ptr<Connection> conn = connectionManager.getConnectionIfAvailable(fd);
-                            if (conn.get() != NULL) {
+                            if (conn.get() != NULL && conn->live) {
```

Comment:
Yes, but I want to wakeup after removing from the list just before I close the connection. By the way, this is only for explicit close of the connection usually during shutdown.